tests/__tests__/sitka_module.test.ts

```diff
@@ -1,98 +1,105 @@
-import { AppState,
-  store,
-  sitka,
-  sitkaNoMiddleware,
-  sitkaWithLogger
+import {
+  sitkaFactory,
+  createSitkaAndStore,
+  AppState,
 } from "../sitka-test"
-
-const { text: textModule } = sitka.getModules()
-const { text: textModuleWithLogger } = sitkaWithLogger.getModules()
-
-describe("Sitka Redux Store", () => {
-  test(`getState returns expected value`, () => {
-    const state: AppState = store.getState()
-
-    const colorActual = state.color
-    const colorExpected = null
-    expect(colorActual).toEqual(colorExpected)
-
-    const textActual = state.text
-    const textExpected = {
-      size: 12,
-      value: "Hello World",
-      numberOfEdits: 0,
-      history: []
-    }
-    expect(textActual).toEqual(textExpected)
-  })
-})
+import { defaultTextModuleState } from "../text_module"
 
 describe("SitkaModule", () => {
   // SETUP
-  let sitkaState
-
-  beforeEach(() => {
-    const modules = sitka.getModules()
-    Object.values(modules).forEach((module: any) => {
-      module.handleReset()
-    })
-
-    sitkaState = store.getState()
-  })
+  const newTextModuleState = {
+    size: 100,
+    value: "test value",
+    numberOfEdits: 1,
+  }
 
   // TESTS
   test('getState returns moduleState', () => {
-    const actual = textModule.getModuleState(sitkaState)
-    const expected = {
-      size: 12,
-      value: "Hello World",
-      numberOfEdits: 0,
-      history: [],
-    }
-
-    expect(actual).toEqual(expected)
+    const { store, sitka } = createSitkaAndStore()
+    const allState = store.getState()
+    const actual = sitka.getModules().text.getStateTestDelegate(allState)
+    expect(actual).toEqual(defaultTextModuleState)
   })
 
   test('mergeState sets partial state of module', () => {
     // handleUpdateSize implements merge state
-    textModule.handleUpdateSize(5)
-    const { text: actual } = store.getState()
+    const { store, sitka } = createSitkaAndStore()
+    sitka.getModules().text.handleUpdateSize(5)
```

Comment:
i don't understand why updating size to 5 will make expected numberOfEdits to 0?

Reply:
tightened scope